src/util/file-filter.js

```diff
@@ -0,0 +1,119 @@
+/* @flow */
+import path from 'path';
+
+import minimatch from 'minimatch';
+
+import {createLogger} from './logger';
+
+const log = createLogger(__filename);
+
+// Use this function to mimic path.resolve without resolving to absolute path.
+export const normalizeResolve = (file: string): string => {
+  // normalize
+  file = path.normalize(file);
+
+  // trim trailing slash
+  if (path.parse(file).base && file.endsWith(path.sep)) {
+    return file.slice(0, -1);
+  }
+  return file;
+};
+
+// FileFilter types and implementation.
+
+export type FileFilterOptions = {|
+  filesToIgnore?: Array<string>,
+  ignoreFiles?: Array<string>,
+  sourceDir?: string,
+  artifactsDir?: string,
+|};
+
+/*
+ * Allows or ignores files.
+ */
+export class FileFilter {
+  filesToIgnore: Array<string>;
+  sourceDir: ?string;
+
+  constructor({
+    filesToIgnore = [
```

Comment:
The `FileFilter` now takes two parameters with two very similar names but they have different meaning, it would be nice to make it more clear by make the names a bit more descriptive (e.g. `filesToIgnore` renamed to `defaultFilesToIgnore`, and `ignoreFiles` renamed to `customFilesToIgnore`?)